spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/BindableProxyFactory.java

```diff
@@ -130,10 +127,12 @@ public void doWith(Method method) throws IllegalArgumentException {
 						MessageChannel sharedChannel = locateSharedChannel(name);
 						if (sharedChannel == null) {
 							inputHolders.put(name, new ChannelHolder(
-									channelFactory.createChannel(name, method.getReturnType()), true));
+									channelFactory.createBindableChannel(name,
+											(Class<? extends MessageChannel>)method.getReturnType()), true));
 						}
 						else {
-							configureSharedMessageChannel(name, method.getReturnType(), sharedChannel);
+							configureSharedMessageChannel(name,
+									(Class<? extends MessageChannel>) method.getReturnType(), sharedChannel);
```

Comment:
maybe move the `method.getReturnType()` call up so it doesn't need to be duplicated